File: 21.调整数组顺序使奇数位于偶数前面/21.调整数组顺序使奇数位于偶数前面.py
# -*- coding:utf-8 -*-
class Solution:
    def reOrderArray(self, array):
        # write code here
        m = len(array)
        k = 0
        for i in range(m):
            if (array[i]%2==1):
                j = i
                while(j>k):
                    array[j],array[j-1] = array[j-1],array[j]
                    j-=1
                k+=1
        return array
            

File: 21.调整数组顺序使奇数位于偶数前面/21.调整数组顺序使奇数位于偶数前面.cpp
class Solution {
public:
    void reOrderArray(vector<int> &array) {
        for(int i = 0 ;i<array.size();i++){
            int key = array[i];
            if (key%2==0){
                continue;
            }
            int j = i - 1;
            while(j>=0 and array[j]%2==0){
                array[j+1] = array[j];
                    j = j - 1;
            }
            array[j+1] = key;
        }
    }
};